### PR TITLE
fix(migrations): Add support for removing imports post migration

### PR DESCRIPTION
**Pull request by jessicajaniuk**
This update removes imports from component decorators and at the top of the files. It also checks to make sure CommonModule is not being used and removes that too.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/schematics/ng-generate/control-flow-migration/util.ts [hunk 1]
```diff
@@ -12,55 +12,94 @@ import ts from 'typescript';
...
-      }
+function analyzeImportDeclarations(
+    node: ts.ImportDeclaration, sourceFile: ts.SourceFile,
+    analyzedFiles: Map<string, AnalyzedFile>) {
+  if (node.getText().indexOf('@angular/common') === -1) {
+    return;
+  }
+  let hasControlFlow = false;
+  const importedItems = node.getChildAt(1).getText();  // the imported items
+  const importList = importedItems.slice(1, importedItems.length - 1).split(',');
```
Comment (crisbeto) [inline packages/core/schematics/ng-generate/control-flow-migration/util.ts:40]: Should we use the AST to remove the elements here to avoid potential string manipulation bugs? The flow is something like this once we get a hold of the `ArrayLiteralExpression`:

```
function removeItem(node: ts.ArrayLiteralExpression) {
  const printer = ts.createPrinter();
  const elements = node.elements.filter(el => ts.isIdentifier(el) && importRemovals.incldues(el.text));
  const updated = ts.factory.updateArrayLiteralExpression(node, elements);

  return printer.printNode(ts.EmitHint.Unspecified, updated, node.getSourceFile());
}
```

The approach is similar for removing from the import declaration.

Comment (jessicajaniuk) [reply packages/core/schematics/ng-generate/control-flow-migration/util.ts:40]: Yeah that seems reasonable to me. I wasn't sure what was the right AST element to use here. So this is very helpful. I'll update it and run the tests.

Comment (JoostK) [reply packages/core/schematics/ng-generate/control-flow-migration/util.ts]: > Yeah that seems reasonable to me. I wasn't sure what was the right AST element to use here. So this is very helpful. I'll update it and run the tests.

@jessicajaniuk  you may be interested to check out https://ts-ast-viewer.com, it shows AST structures and factory snippets.

File: packages/core/schematics/ng-generate/control-flow-migration/index.ts [hunk 2]
```diff
@@ -76,17 +76,17 @@ function runControlFlowMigration(
     analyze(sourceFile, analysis);
   }
 
-  for (const [path, file] of analysis) {
+  for (let [path, file] of analysis) {
```
Comment (crisbeto) [inline packages/core/schematics/ng-generate/control-flow-migration/index.ts:79]: This change seems unrelated since I don't see where these variables are being changed.

**Review by crisbeto — APPROVED** [review]

File: packages/core/schematics/ng-generate/control-flow-migration/index.ts (hunk 2 above)
Comment (jessicajaniuk) [reply packages/core/schematics/ng-generate/control-flow-migration/index.ts:79]: Yeah I think that was a refactor that's no longer relevant. I'll update it.

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 17adf0f917bb30f24255a0dea1c25e40ec619bf4.
